### DISCUSSION TITLE
Are there any tools or techniques that can analyze NUMA(Non-Uniform Memory Access) problems?

### DISCUSSION THREAD

**Discussion by armorhu** [skypjack/entt#816] [General]
When we used ecs at that time, we always wanted to know how to plan components and system is the best choice, and the measurement standard is continuous access to memory.
So I would like to ask if there are any tools or techniques that can quantify this indicator.

For example: you can output all the memory access addresses of a piece of code and draw them into a chart, similar to this:
![image](https://user-images.githubusercontent.com/4246258/147462732-24ad9747-c3ca-4abf-a225-5c39a04e876c.png)

Or output the number of cache-misses during execution。

**Comment by skypjack** [comment]
I think both intel vtune and perf if you want a free tool can give you a grasp on the cache misses during execution.
